cbits/runProcess.c

```diff
@@ -182,6 +185,22 @@ runInteractiveProcess (char *const args[],
         }
 
         if ( childUser) {
+            // Using setuid properly first requires that we initgroups.
+            // However, to do this we must know the username of the user we are
+            // switching to.
+            struct passwd pw;
+            struct passwd *res = NULL;
+            int buf_len = sysconf(_SC_GETPW_R_SIZE_MAX);
+            char *buf = malloc(buf_len);
+            gid_t suppl_gid = childGroup ? *childGroup : getgid();
+            if ( getpwuid_r(*childUser, &pw, buf, buf_len, &res) != 0) {
+                childFailed(forkCommunicationFds[1], forkGetpwuidFailed);
+            } else if (res == NULL) {
```

Comment:
The mixing of `else if` and `if` makes the code confusing, at least IMO. `childFailed` will cause the program to exit regardless, so the `else` isn't strictly necessary, but it arguably helps with readability. I don't care too much one direction or the other, but I'd prefer to see the code be consistent.

If there's a reason for using `else if` in just one of the checks, please clarify, ideally with an inline comment.

Reply:
Indeed looking back at this I can't imagine what I was thinking. This `else if` looks just plain wrong.

Reply:
Ahh, no, it's not wrong but indeed the `else` isn't necessary. Fixed.